### DISCUSSION TITLE
Longest Substring Without Repeating Characters

### DISCUSSION THREAD

**Discussion by ghost** [community/community#138505] [Programming Help] [answered]
### Body

Write a function length_of_longest_substring that takes a string as input and returns the length of the longest substring without repeating characters.

**_Requirements:_**
Function Signature:

`def length_of_longest_substring(s: str) -> int:`

Example:

`length_of_longest_substring("abcabcbb")  # Output: 3 (The longest substring without repeating characters is "abc")`
`length_of_longest_substring("bbbbb")     # Output: 1 (The longest substring without repeating characters is "b")`
`length_of_longest_substring("pwwkew")    # Output: 3 (The longest substring without repeating characters is "wke")`

**_Constraints:_**

The input string s consists of English letters, digits, symbols, and spaces.
The length of the input string will be at most 10^5.

**_Approach:_**

Consider using the sliding window technique along with a set to keep track of characters in the current window:

1. Initialize two pointers, start and end, both set to the beginning of the string.
2. Use a set to track characters in the current window.
3. Expand the end pointer to include new characters in the window while ensuring no duplicates are present.
4. If a duplicate character is found, move the start pointer to the right until the duplicate is removed.
5. Keep track of the maximum length of the window during the process.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by MatheusFigueiredo7** [comment, marked as answer]
Think that can solve it

```
def length_of_longest_substring(s: str) -> int:
    char_set = set()
    max_length = 0
    start = 0
    
    for end in range(len(s)):
        while s[end] in char_set:
            char_set.remove(s[start])
            start += 1
        char_set.add(s[end])
        max_length = max(max_length, end - start + 1)
    
    return max_length
```

Peace
